sys/linux/smack.txt

```diff
@@ -1,21 +1,213 @@
+#type smack_perm stringnoz
+type smack_perm stringnoz[perm]
+
+smackfs_access {
+	subject	smack_labelnoz
+	sp0	const[' ', int8]
+	object	smack_labelnoz
+	sp1	const[' ', int8]
+	access	smack_perm
+	z	const[0, int8]
+} [packed]
+
+smackfs_change_access {
+	subject	smack_labelnoz
+	sp0	const[' ', int8]
+	object	smack_labelnoz
+	sp1	const[' ', int8]
+	allow	smack_perm
+	sp2	const[' ', int8]
+	deny	smack_perm
+	z	const[0, int8]
+} [packed]
+
+resource fd_smackfs_access[fd]
+smackfs_check_access = "/sys/fs/smackfs/access", "/sys/fs/smackfs/access2"
+openat$smackfs_access(fd const[AT_FDCWD], file ptr[in, string[smackfs_check_access]], flags const[O_RDWR], mode const[0]) fd_smackfs_access
+write$smackfs_access(fd fd_smackfs_access, buf ptr[in, smackfs_access], count len[buf])
+read$smackfs_access(fd fd_smackfs_access, buf ptr[out, fmt[dec, int32[0:1]]], count len[buf])
+
+# load, load2, load-self, load-self2
+resource fd_smackfs_load[fd]
+smackfs_load_files = "/sys/fs/smackfs/load", "/sys/fs/smackfs/load2", "/sys/fs/smackfs/load-self", "/sys/fs/smackfs/load-self2"
+openat$smackfs_load(fd const[AT_FDCWD], file ptr[in, string[smackfs_load_files]], flags const[O_RDWR], mode const[0]) fd_smackfs_load
+write$smackfs_load(fd fd_smackfs_load, buf ptr[in, smackfs_access], count len[buf])
+
+# ambient, unconfined, syslog, revoke-subject
+resource fd_smackfs_label[fd]
+smackfs_label_files = "/sys/fs/smackfs/ambient", "/sys/fs/smackfs/syslog", "/sys/fs/smackfs/unconfined", "/sys/fs/smackfs/revoke-subject"
+openat$smackfs_label(fd const[AT_FDCWD], file ptr[in, string[smackfs_label_files]], flags const[O_RDWR], mode const[0]) fd_smackfs_label
+write$smackfs_label(fd fd_smackfs_label, buf ptr[in, smack_label], count len[buf])
+
+resource fd_smackfs_change_rule[fd]
+openat$smackfs_change_rule(fd const[AT_FDCWD], file ptr[in, string["/sys/fs/smackfs/change-rule"]], flags const[O_RDWR], mode const[0]) fd_smackfs_change_rule
+write$smackfs_change_rule(fd fd_smackfs_change_rule, buf ptr[in, smackfs_change_access], count len[buf])
+
+smackfs_cipso_category {
+	cat	fmt[dec, int8[0:SMACK_CIPSO_MAXCATNUM]]
+	sp	const[' ', int8]
+}
+
+smackfs_cipso {
+	label	smack_labelnoz
+	sp0	const[' ', int8]
+# SMK_DIGITLEN == 4
```

Comment:
What exactly does this mean?

Reply:
I will replace it with this one:
```
# int8 because of SMK_DIGITLEN == 4
```
If you look at the code, you could note that it uses sscanf "%d" for reading numbers. But we should pin there numbers to int8 because the format is fixed and next number should be +SMK_DIGITLEN from previous one.
https://elixir.bootlin.com/linux/v5.4-rc8/source/security/smack/smackfs.c#L885